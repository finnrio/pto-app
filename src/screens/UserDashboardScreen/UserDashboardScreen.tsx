import React from "react";
import { Text, TouchableOpacity, View } from "react-native";
import styles from "./styles";

export default function UserDashboardScreen({ navigation }: any) {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>
        Welcome to the employee user dashboard. Here you can create and query
        PTO requests and events.
      </Text>
      <TouchableOpacity
        style={styles.button}
        onPress={() => navigation.navigate("My Activity")}
      >
        <Text style={styles.buttonTitle}>My Activity</Text>
      </TouchableOpacity>

      <TouchableOpacity
        style={styles.button}
        onPress={() => navigation.navigate("Calendar")}
      >
        <Text style={styles.buttonTitle}>Calendar</Text>
      </TouchableOpacity>
    </View>
  );
}
