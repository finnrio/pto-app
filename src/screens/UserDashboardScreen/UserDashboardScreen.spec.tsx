import {
  cleanup,
  fireEvent,
  render,
  screen,
} from "@testing-library/react-native";
import React from "react";
import UserDashboardScreen from "./UserDashboardScreen";

const navigation = {
  navigate: jest.fn(),
};

describe("UserDashboardScreen", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });
  it("matches the snapshot", () => {
    const tree = render(<UserDashboardScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe("My Activity Button", () => {
    it("takes the user to the 'My Activity' screen", () => {
      render(<UserDashboardScreen navigation={navigation} />);
      fireEvent.press(screen.getByTestId("my_activity_btn"));
      expect(navigation.navigate).toHaveBeenCalledWith("My Activity");
    });
  });
  describe("Calendar Button", () => {
    it("takes the user to the 'Calendar' screen", () => {
      render(<UserDashboardScreen navigation={navigation} />);
      fireEvent.press(screen.getByTestId("calendar_btn"));
      expect(navigation.navigate).toHaveBeenCalledWith("Calendar");
    });
  });
});
