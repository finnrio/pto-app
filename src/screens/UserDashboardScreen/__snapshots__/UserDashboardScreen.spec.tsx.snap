// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDashboardScreen matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 10,
        "paddingLeft": 16,
      }
    }
  >
    Welcome to the employee user dashboard. Here you can create and query PTO requests and events.
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#788eec",
        "borderRadius": 5,
        "height": 48,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "opacity": 1,
      }
    }
    testID="my_activity_btn"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      My Activity
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#788eec",
        "borderRadius": 5,
        "height": 48,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
        "opacity": 1,
      }
    }
    testID="calendar_btn"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      Calendar
    </Text>
  </View>
</View>
`;
